bl.js

```diff
@@ -57,8 +57,14 @@ BufferList.prototype.append = function (buf) {
   if (typeof buf == 'number')
     buf = buf.toString()
 
-  this._bufs.push(isBuffer ? buf : new Buffer(buf))
-  this.length += buf.length
+  if (buf instanceof BufferList) {
+    this._bufs = this._bufs.concat(buf._bufs)
```

Comment:
Should we do `this._bufs.push.apply(this._bufs, buf._bufs)` to prevent creation of a new array?